hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/utils/HddsVolumeUtil.java

```
@@ -122,4 +129,28 @@ private static void mapDbVolumesToDataVolumesIfNeeded(
         hddsVolume.setDbVolume(globalDbVolumeMap.getOrDefault(
             hddsVolume.getStorageID(), null)));
   }
+
+  /**
+   * Get the HddsVolume according to the path.
+   * @param volumes volume list to match from
+   * @param pathStr path to match
+   */
+  public static HddsVolume matchHddsVolume(List<HddsVolume> volumes,
+      String pathStr) {
+    assert pathStr != null;
+    List<HddsVolume> resList = new ArrayList<>();
+    for (HddsVolume hddsVolume: volumes) {
+      if (pathStr.startsWith(hddsVolume.getVolumeRootDir())) {
+        resList.add(hddsVolume);
+      }
+    }
+    if (resList.size() == 1) {
+      return resList.get(0);
+    } else if (resList.size() > 1) {
+      LOG.warn("Get multi volumes {} matching path {}",
+          resList.stream().map(StorageVolume::getVolumeRootDir).collect(
+              Collectors.joining(",")), pathStr);
```

I think it is better to throw an exception in this case.